### ISSUE TITLE
scipy build failure wrt changes in r7712 of numpy (migrated from Trac #1294)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2847]
Original ticket http://projects.scipy.org/numpy/ticket/1294
Reported 2009-11-09 by atmention:nilswagner01, assigned to unknown.

The problem file is generated by f2py.
    In file included from scipy/integrate/_odepackmodule.c:6:
    scipy/integrate/__odepack.h: In function ‘odepack_odeint’:
    scipy/integrate/__odepack.h:328: warning: call to function
    ‘lsoda_’ without a real prototype
    scipy/integrate/__odepack.h:31: note: ‘lsoda_’ was
    declared here
    scipy/integrate/_odepackmodule.c: At top level:
    scipy/integrate/multipack.h:114: warning:
    ‘my_make_numpy_array’ defined but not used
    scipy/integrate/__odepack.h: In function ‘odepack_odeint’:
    scipy/integrate/__odepack.h:226: warning: ‘tcrit’ may be
    used uninitialized in this function
    scipy/integrate/__odepack.h:227: warning: ‘wa’ may be used
    uninitialized in this function
    /usr/bin/gfortran -Wall -Wall -shared
    build/temp.linux-x86_64-2.6/scipy/integrate/_odepackmodule.o
    -L/home/nwagner/src/ATLAS3.8.2/mybuild/lib -L/usr/lib64
    -Lbuild/temp.linux-x86_64-2.6 -lodepack -llinpack_lite
    -lmach -lptf77blas -lptcblas -latlas -lpython2.6
    -lgfortran -o
    build/lib.linux-x86_64-2.6/scipy/integrate/_odepack.so
    building 'scipy.integrate.vode' extension
    compiling C sources
    C compiler: /usr/bin/gcc -fno-strict-aliasing -DNDEBUG
    -fmessage-length=0 -O2 -Wall -D_FORTIFY_SOURCE=2
    -fstack-protector -funwind-tables
    -fasynchronous-unwind-tables -g -fwrapv -fPIC

```
creating
build/temp.linux-x86_64-2.6/build/src.linux-x86_64-2.6/scipy/integrate
compile options: '-DATLAS_INFO="\"3.8.2\""
-I/home/nwagner/src/ATLAS3.8.2/include
-Ibuild/src.linux-x86_64-2.6
-I/home/nwagner/local/lib64/python2.6/site-packages/numpy/core/include
-I/usr/include/python2.6 -c'
gcc:
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:
In function ‘cb_f_in_dvode__user__routines’:
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:376:
error: ‘y_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:376:
error: (Each undeclared identifier is reported only once
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:376:
error: for each function it appears in.)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:376:
error: ‘n’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:377:
error: ‘ydot_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:335:
warning: unused variable ‘ipar_5’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:334:
warning: unused variable ‘rpar_4’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:332:
warning: unused variable ‘n_0’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:
In function ‘cb_jac_in_dvode__user__routines’:
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:508:
error: ‘y_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:508:
error: ‘n’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:509:
error: ‘jac_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:509:
error: ‘nrowpd’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:509:
warning: left-hand operand of comma expression has no
effect
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:467:
warning: unused variable ‘ipar_8’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:466:
warning: unused variable ‘rpar_7’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:465:
warning: unused variable ‘nrowpd_6’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:464:
warning: unused variable ‘mu_4’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:463:
warning: unused variable ‘ml_3’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:461:
warning: unused variable ‘n_0’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:
In function ‘cb_f_in_zvode__user__routines’:
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:636:
error: ‘y_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:636:
error: ‘n’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:637:
error: ‘ydot_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:595:
warning: unused variable ‘ipar_5’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:594:
warning: unused variable ‘rpar_4’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:592:
warning: unused variable ‘n_0’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:
In function ‘cb_jac_in_zvode__user__routines’:
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:768:
error: ‘y_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:768:
error: ‘n’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:769:
error: ‘jac_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:769:
error: ‘nrowpd’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:769:
warning: left-hand operand of comma expression has no
effect
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:727:
warning: unused variable ‘ipar_8’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:726:
warning: unused variable ‘rpar_7’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:725:
warning: unused variable ‘nrowpd_6’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:724:
warning: unused variable ‘mu_4’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:723:
warning: unused variable ‘ml_3’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:721:
warning: unused variable ‘n_0’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:
In function ‘cb_f_in_dvode__user__routines’:
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:376:
error: ‘y_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:376:
error: (Each undeclared identifier is reported only once
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:376:
error: for each function it appears in.)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:376:
error: ‘n’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:377:
error: ‘ydot_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:335:
warning: unused variable ‘ipar_5’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:334:
warning: unused variable ‘rpar_4’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:332:
warning: unused variable ‘n_0’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:
In function ‘cb_jac_in_dvode__user__routines’:
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:508:
error: ‘y_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:508:
error: ‘n’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:509:
error: ‘jac_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:509:
error: ‘nrowpd’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:509:
warning: left-hand operand of comma expression has no
effect
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:467:
warning: unused variable ‘ipar_8’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:466:
warning: unused variable ‘rpar_7’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:465:
warning: unused variable ‘nrowpd_6’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:464:
warning: unused variable ‘mu_4’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:463:
warning: unused variable ‘ml_3’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:461:
warning: unused variable ‘n_0’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:
In function ‘cb_f_in_zvode__user__routines’:
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:636:
error: ‘y_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:636:
error: ‘n’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:637:
error: ‘ydot_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:595:
warning: unused variable ‘ipar_5’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:594:
warning: unused variable ‘rpar_4’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:592:
warning: unused variable ‘n_0’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:
In function ‘cb_jac_in_zvode__user__routines’:
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:768:
error: ‘y_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:768:
error: ‘n’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:769:
error: ‘jac_Dims’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:769:
error: ‘nrowpd’ undeclared (first use in this function)
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:769:
warning: left-hand operand of comma expression has no
effect
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:727:
warning: unused variable ‘ipar_8’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:726:
warning: unused variable ‘rpar_7’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:725:
warning: unused variable ‘nrowpd_6’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:724:
warning: unused variable ‘mu_4’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:723:
warning: unused variable ‘ml_3’
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c:721:
warning: unused variable ‘n_0’
error: Command "/usr/bin/gcc -fno-strict-aliasing -DNDEBUG
-fmessage-length=0 -O2 -Wall -D_FORTIFY_SOURCE=2
-fstack-protector -funwind-tables
-fasynchronous-unwind-tables -g -fwrapv -fPIC
-DATLAS_INFO="\"3.8.2\""
-I/home/nwagner/src/ATLAS3.8.2/include
-Ibuild/src.linux-x86_64-2.6
-I/home/nwagner/local/lib64/python2.6/site-packages/numpy/core/include
-I/usr/include/python2.6 -c
build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.c -o
build/temp.linux-x86_64-2.6/build/src.linux-x86_64-2.6/scipy/integrate/vodemodule.o"
failed with exit status 1
```

**Comment by thouis** [comment]
Comment in Trac by atmention:nilswagner01, 2009-11-09

Duplicate of [http://projects.scipy.org/numpy/ticket/1293]

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2009-11-10
